staging/src/k8s.io/apimachinery/pkg/util/net/interface.go

```diff
@@ -36,53 +33,57 @@ type Route struct {
 	// TODO: add more fields here if needed
 }
 
-func getRoutes(input io.Reader) ([]Route, error) {
-	routes := []Route{}
-	if input == nil {
-		return nil, fmt.Errorf("input is nil")
-	}
-	scanner := bufio.NewReader(input)
-	for {
-		line, err := scanner.ReadString('\n')
-		if err == io.EOF {
-			break
-		}
-		//ignore the headers in the route info
-		if strings.HasPrefix(line, "Iface") {
-			continue
-		}
-		fields := strings.Fields(line)
-		routes = append(routes, Route{})
-		route := &routes[len(routes)-1]
-		route.Interface = fields[0]
-		ip, err := parseIP(fields[1])
-		if err != nil {
-			return nil, err
-		}
-		route.Destination = ip
-		ip, err = parseIP(fields[2])
-		if err != nil {
-			return nil, err
-		}
-		route.Gateway = ip
+const (
+	ipCommand   = "/usr/sbin/ip"
+	grepCommand = "/usr/bin/grep"
+)
+
+var (
+	ipv4_zero = net.IP{0, 0, 0, 0}
+	ipv6_zero = net.ParseIP("::")
+)
+
+var execCommand = exec.Command
+
+// buildRouteCommand forms IP route command(s) to obtain routes and filter
+// on the default routes. Can do IPv4 only, or IPv4 + IPv6.
+func buildRouteCommand(withIPv6 bool) string {
```

Comment:
I elaborated on my questions, in the issue for this commit (44848). Thought maybe it would be best to discuss them there, rather than under the review comments, as they will have a major impact on the implementation.